device.go

```diff
@@ -78,7 +80,18 @@ func (d *device) InstallUpdate(image io.ReadCloser, size int64) error {
 		return err
 	}
 
-	b := &BlockDevice{Path: inactivePartition}
+	// This condition is to handle UBI block devices. Which is not prefixed
+	// with /dev due to the fact that the kernel root= argument does not
+	// handle UBI block devices which are prefixed with /dev
+	//
+	// Kernel root= only accepts:
+	// - ubi0_0
+	// - ubi:rootfsa
+	if strings.HasPrefix(inactivePartition, "/dev/") == false {
+		inactivePartition = path.Join("/dev", inactivePartition)
```

Comment:
I think it might be safer to use `filepath` instead of `path`. This will remove some OS dependency if will be needed in future.